### PR TITLE
Fixed #27656 -- Updated docstring verbs according to PEP 257.

### PR DESCRIPTION
**Pull request by desecho**
https://code.djangoproject.com/ticket/27656

### CODE REVIEW

File: django/test/client.py [hunk 1]
```diff
@@ -462,9 +462,9 @@ def session(self):
 
     def request(self, **request):
         """
-        The master request method. Composes the environment dictionary
-        and passes to the handler, returning the result of the handler.
-        Assumes defaults for the query environment, which can be overridden
+        The master request method. Compose the environment dictionary
+        and pass to the handler, return the result of the handler.
```
Comment (timgraham) [inline django/test/client.py]: , -> , and

File: django/test/client.py [hunk 2]
```diff
@@ -674,9 +674,9 @@ def _login(self, user, backend=None):
 
     def logout(self):
         """
-        Removes the authenticated user's cookies and session object.
+        Remove the authenticated user's cookies and session object.
```
Comment (timgraham) [inline django/test/client.py]: "Log out the user by removing the cookies and session object."
(remove second sentence)

File: django/test/client.py [hunk 3]
```diff
@@ -701,7 +701,7 @@ def _parse_json(self, response, **extra):
         return response._json
 
     def _handle_redirects(self, response, **extra):
-        "Follows any redirects by requesting responses from the server using GET."
+        "Follow any redirects by requesting responses from the server using GET."
```
Comment (timgraham) [inline django/test/client.py]: Pep257 says, "Triple quotes are used even though the string fits on one line. This makes it easy to later expand it."

File: django/test/client.py [hunk 4]
```diff
@@ -441,14 +441,14 @@ def __init__(self, enforce_csrf_checks=False, **defaults):
...
     def store_exc_info(self, **kwargs):
         """
-        Stores exceptions when they are generated by a view.
+        Store exceptions when they are generated by a view.
         """
         self.exc_info = sys.exc_info()
 
     @property
     def session(self):
         """
```
Comment (timgraham) [inline django/test/client.py]: Should these by one liners ? e.g. """Obtain the current session variables.""" -- we currently don't have consistency about that.

**Comment by desecho** [comment]
@timgraham I'll be making the changes you suggested as well as fixes for the verb style. Also, I noticed that this coding style is not being followed as well. I should probably be fixing that as well - I go through the all of the comments anyway. 

> In test docstrings, state the expected behavior that each test demonstrates. Don’t include preambles such as “Tests that” or “Ensures that”.

And I assume it is a good idea to also make sure these PEP rules are being followed as well:

- Triple quotes are used even though the string fits on one line. This makes it easy to later expand it.
- The closing quotes are on the same line as the opening quotes. This looks better for one-liners.
- There's no blank line either before or after the docstring. (except for classes)
- The docstring is a phrase ending in a period.

**Comment by timgraham** [comment]
Keep this patch focused on django/ only -- don't worry about test docstrings, there's a separate ticket for that.
The PEP rules you mentioned are fine.

**Comment by desecho** [comment]
Ok. Also I noticed that the wrapping is sometimes different. It can be 72 or less than 79. Since you mentioned it in the commit we should probably make it consistent to 79.
> Documentation, comments, and docstrings should be wrapped at 79 characters, even though PEP 8 suggests 72.

**Comment by timgraham** [comment]
If you're making significant changes to the docstring, feel free to adjust the length to 79, otherwise don't make needless changes.

**Comment by desecho** [comment]
I made some changes. Please see if I am on the right track.

File: tests/admin_checks/tests.py [hunk 5]
```diff
@@ -158,7 +158,7 @@ class SongAdmin(admin.ModelAdmin):
 
     def test_custom_modelforms_with_fields_fieldsets(self):
         """
-        # Regression test for #8027: custom ModelForms with fields/fieldsets
```
Comment (timgraham) [inline tests/admin_checks/tests.py]: As I mentioned before, please don't modify any test docstrings for this PR.

File: django/test/client.py [hunk 6]
```diff
@@ -39,9 +39,8 @@
 
 
 class RedirectCycleError(Exception):
-    """
-    The test client has been asked to follow a redirect loop.
-    """
+    """The test client has been asked to follow a redirect loop."""
+
```
Comment (timgraham) [inline django/test/client.py]: I'd skip the blank line after the class docstring changes. Focus on changing the verbs only.

File: django/contrib/admin/templatetags/admin_list.py [hunk 7]
```diff
@@ -298,10 +288,13 @@ def link_in_col(is_first, field_name, cl):
 
 
 class ResultList(list):
-    # Wrapper class used to return items in a list_editable
-    # changelist, annotated with the form object for error
-    # reporting purposes. Needed to maintain backwards
-    # compatibility with existing admin templates.
+    """
+    Wrapper class used to return items in a list_editable
```
Comment (timgraham) [inline django/contrib/admin/templatetags/admin_list.py]: Might as well reflow the line length in a case like this.

File: django/contrib/admin/checks.py [hunk 8]
```diff
@@ -420,8 +419,8 @@ def _check_prepopulated_fields_value(self, obj, model, val, label):
             ]))
 
     def _check_prepopulated_fields_value_item(self, obj, model, field_name, label):
-        """ For `prepopulated_fields` equal to {"slug": ("title",)},
-        `field_name` is "title". """
+        """For `prepopulated_fields` equal to {"slug": ("title",)},
```
Comment (timgraham) [inline django/contrib/admin/checks.py]: Be consistent about the multiline style -- use
```
"""
....
"""
```

File: django/apps/registry.py [hunk 9]
```diff
@@ -205,9 +199,11 @@ def get_model(self, app_label, model_name=None, require_ready=True):
         return app_config.get_model(model_name, require_ready=require_ready)
 
     def register_model(self, app_label, model):
-        # Since this method is called when models are imported, it cannot
-        # perform imports because of the risk of import loops. It mustn't
-        # call get_app_config().
+        """
+        Since this method is called when models are imported, it cannot
```
Comment (timgraham) [inline django/apps/registry.py]: This content isn't a docstring, so inline comment is more appropriate.

File: django/apps/config.py [hunk 10]
```diff
@@ -9,9 +9,7 @@
 
 
 class AppConfig(object):
-    """
-    Class representing a Django application and its configuration.
-    """
+    """Class representing a Django application and its configuration."""
```
Comment (timgraham) [inline django/apps/config.py]: If there aren't any content changes required, don't bother changing a docstring like this.

**Comment by timgraham** [comment]
Please do only verb changes in this PR. I think all other changes could be refactored automatically with some tool if we decide to do that. Otherwise, this is just creating a huge unreviewable patch.

**Comment by desecho** [comment]
So I assume this means don't follow your previous suggestions and do only verb changes. I caught few typos - I will include them as well.

**Comment by timgraham** [comment]
Yes, sorry I had second thoughts about this when I started seeing how many more changes would be required as well as the thought that other changes could be automated.

**Comment by timgraham** [comment]
On further thought, I'd hold off on doing further changes for a couple weeks so that you don't create merge conflicts with all that code that will be removed in Django 2.0 as part of the ending deprecations (https://github.com/django/django/pull/7773). I'll leave a note when it's a good time to continue. Thanks!

**Comment by desecho** [comment]
I guess we can continue with this ticket since #7773 got merged.

**Comment by timgraham** [comment]
Sure, feel free to rebase. Maybe you can submit some smaller commits for easier review, such as only doing "django.contrib" so that the patch doesn't get so long that GitHub starts displaying "Load diff" buttons. It gets tedious to click those to review things.

**Comment by desecho** [comment]
I completed all of the changes.

**Comment by timgraham** [comment]
What was your methodology for identifying places that need to be fixed? Working my way through `contrib.admin` (so don't bother reviewing that one) I see a lot of places where, for example, the first verb in the sentence was fixed but not the second (e.g. "Remove the authenticated user's ID from the request and flushes their". Even grepping for "Returns" shows places that weren't updated. Another phrasing to check for is something like "If no user is retrieved an instance of `AnonymousUser` is returned." which should be "If no user is retrieved, return an instance of `AnonymousUser`."  I know this is really tedious, but reviewing a partially completed attempt at this isn't fun either.

**Comment by timgraham** [comment]
django.contrib commit merged in 5411821e3b8d1427ee63a5914aed1088c04cc1ed (with many edits).

**Pull request merged**